### ISSUE TITLE
Rust syntax issues 

### ISSUE THREAD

**Issue by dten** [sublimehq/Packages#251]
I've pulled the latest changes and have some other issues :)

```
    fn collect_vec() {
        let _: Vec<(usize, usize)> = (0..10).enumerate().collect::<Vec<_>>();
        let _: Vec<(usize, usize)> = vec!();
        let _: Vec<(usize, usize)> = vec!{};
        let _: Vec<(usize, usize)> = vec![];
    }
```
the opening bracket of `(usize, usize)` and the `_` in the `Vector` 's item type aren't happy and come out bright pink
also the 3 ways of usages of macros come out differently highlighted (not sure it's intentional)

```
macro_rules! forward_ref_binop {
    (impl $imp:ident, $method:ident for $t:ty, $u:ty) => {
        impl<'a, 'b> $imp<&'a $u> for &'b $t {
            type Output = <$t as $imp<$u>>::Output;

            #[inline]
            fn $method(self, other: &'a $u) -> <$t as $imp<$u>>::Output {
                $imp::$method(*self, *other)
            }
        }
    }
}
```

the various templated items stuggle to be highlighted the same

**Comment by wbond** [comment]
Ah yes, I support the `$metavariables` can be anywhere.

Do you know if there is any other syntax that uses `$` in Rust?

**Comment by dten** [comment]
https://doc.rust-lang.org/book/syntax-index.html suggests there are 3 macro related usages. I've never used the third one though.

**Comment by dten** [comment]
Another one, `=` not happy
```
pub fn pow<T: One + Clone>(mut base: T, mut exp: usize) -> T
        where for <'a> &'a T: Mul<&'a T,Output = T>, {

    unimplemented!();
}
```

**Comment by wbond** [comment]
These issues should be resolved by 112f7906615878ad9168b1d9dcc9a14f3179e398

**Comment by dten** [comment]
Thanks
